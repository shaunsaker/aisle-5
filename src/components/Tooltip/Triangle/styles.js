import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  container: {
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent',
    borderBottomColor: styleConstants.colors.secondary,
  },
});

export default styles;
