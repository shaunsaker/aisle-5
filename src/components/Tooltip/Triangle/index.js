import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const propTypes = {
  height: PropTypes.number,
};

const defaultProps = {};

const Triangle = ({ height }) => {
  return (
    <View
      style={[
        styles.container,
        { borderLeftWidth: height, borderRightWidth: height, borderBottomWidth: height },
      ]}
    />
  );
};

Triangle.propTypes = propTypes;
Triangle.defaultProps = defaultProps;

export default Triangle;
