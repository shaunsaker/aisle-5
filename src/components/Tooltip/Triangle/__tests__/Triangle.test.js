import React from 'react';
import renderer from 'react-test-renderer';

import Triangle from '..';

describe('Triangle', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Triangle />);

      expect(component).toMatchSnapshot();
    });
  });
});
