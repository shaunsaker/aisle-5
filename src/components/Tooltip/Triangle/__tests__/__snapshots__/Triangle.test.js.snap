// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Triangle renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#FEA116",
        "borderLeftColor": "transparent",
        "borderRightColor": "transparent",
      },
      Object {
        "borderBottomWidth": undefined,
        "borderLeftWidth": undefined,
        "borderRightWidth": undefined,
      },
    ]
  }
/>
`;
